src/main/java/org/folio/fqm/utils/EntityTypeUtils.java

```diff
@@ -95,4 +98,42 @@ public static List<String> getDateFields(EntityType entityType) {
       .map(org.folio.querytool.domain.dto.Field::getName)
       .toList();
   }
+
+  /**
+   * Searches for a column within an entity type by name, returning it if it exists.
+   * This method will not search nested object fields, only top-level columns.
+   */
+  public static Optional<EntityTypeColumn> findColumnByName(EntityType entityType, String columnName) {
+    return entityType.getColumns().stream().filter(column -> column.getName().equals(columnName)).findFirst();
+  }
+
+  /**
+   * Searches for a join from source to target and returns it, if it exists.
+   * The join MUST be defined in source; a separate call is needed to check for joins defined target to source.
+   */
+  public static Optional<Join> findJoinBetween(EntityTypeColumn source, EntityTypeColumn target) {
+    return source
+      .getJoinsTo()
+      .stream()
+      .filter(j ->
+        j.getTargetId().equals(target.getOriginalEntityTypeId()) &&
+        j.getTargetField().equals(splitFieldIntoAliasAndField(target.getName()).getRight())
+      )
+      .findFirst();
+  }
+
+  /**
+   * Splits a composite field name (foo.bar) into an alias and a field name.
+   * 
+   * @example foo.bar -> Pair.of("foo", "bar")
+   * @example bar -> Pair.of("", "bar")
+   * @example foo.bar.baz -> Pair.of("foo.bar", "baz")
+   */
+  public static Pair<String, String> splitFieldIntoAliasAndField(String field) {
+    int dotIndex = field.lastIndexOf('.');
+    if (dotIndex == -1) {
+      return Pair.of("", field);
+    }
+    return Pair.of(field.substring(0, dotIndex), field.substring(dotIndex + 1));
```

Comment:
You could shrink this down to a one-liner pretty easily with `String.split()`, rather than having to roll your own version. All you have to do is reverse the field name, split once on "\\.", then reverse the resulting array and array contents 🙂 

```suggestion
return Optional.ofNullable(StreamSupport.stream(Spliterators.spliterator(Arrays.stream(new StringBuilder("blah.bleh.blih").reverse().toString().split("\\.", 2)).map(StringBuilder::new).map(StringBuilder::reverse).collect(Collectors.toCollection(LinkedList::new)).descendingIterator(), 2, Spliterator.ORDERED), false).toList()).map(p -> Pair.of(p.get(0), p.get(1))).orElse(null);
```

Simple!

Reply:
![stop it, get some help gif](https://media.tenor.com/mZZoOtDcouoAAAAM/stop-it-get-some-help.gif)